rustls-platform-verifier/src/verification/android.rs

```diff
@@ -77,21 +77,42 @@ impl Verifier {
         Self {
             #[cfg(any(test, feature = "ffi-testing"))]
             test_only_root_ca_override: None,
-            default_provider: OnceCell::new(),
+            crypto_provider: OnceCell::new(),
         }
     }
 
+    /// Attempts to set the [`CryptoProvider`] for this `Verifier`.
+    ///
+    /// # Panics
+    ///
+    /// This function will panic if this verifier's crypto provider has already been set.
+    pub fn set_provider(&mut self, crypto_provider: Arc<CryptoProvider>) {
+        self.crypto_provider
```

Comment:
Why not propagate the error instead, letting the caller chose how to handle it?

Reply:
I agree

Reply:
I realized we have `&mut` here so it's infallible :tada: